### ISSUE TITLE
Merge query optimizer

### ISSUE THREAD

**Issue by jiashenC** [georgia-tech-db/eva#135]

**Comment by xzdandy** [comment]
New Operator Support in Optimizer:
- [ ] Order By
- [ ] Limit

**Comment by xzdandy** [comment]
Unittest
- [ ] binder
- [ ] group (Current group only support one logic expression)
- [ ] memo
- [ ] optimizer_context (xform function)
- [ ] optimizer_task (TopDown, BottomUp and OptimizeInputs)
- [ ] New plan_generator
- [ ] mat operator

We also probably need unittest for every rewrite rule, what's your opinion @gaurav274?

**Comment by jiashenC** [comment]
I add unittests to cover all rewrite rules for now. Those implementations rules seem very trivial. 

- [ ] Rule

**Comment by jiashenC** [comment]
Just double check union is also missing in optimizer right? Or you do not want to support union for now?

**Comment by xzdandy** [comment]
I think Union is supported.  https://github.com/georgia-tech-db/eva/blob/c2ae388eb83ac1f83fba9b0359cc8a857699d507/src/optimizer/rules/rules.py#L507

**Comment by jiashenC** [comment]
Yes, it has a rule, but no generator. https://github.com/georgia-tech-db/eva/tree/optimizer-merge/src/optimizer/generators

**Comment by xzdandy** [comment]
It is in the seq scan.

**Comment by jiashenC** [comment]
Ah, i see. I completely miss that part.
